win32/src/socket_transport_win32.c

```diff
@@ -0,0 +1,720 @@
+            }
+        }
+        closesocket(client_socket);
+    }
+all_ok:
+    return result;
+}
+
+SOCKET_TRANSPORT_HANDLE socket_transport_create(SOCKET_TYPE type)
+{
+    SOCKET_TRANSPORT* result;
+
+    // Codes_SOCKET_TRANSPORT_WIN32_09_001: [ socket_transport_create shall ensure type is either SOCKET_CLIENT, or SOCKET_SERVER. ]
+    if (type != SOCKET_CLIENT && type != SOCKET_SERVER)
+    {
+        LogError("Invalid socket type specified: %" PRI_MU_ENUM "", MU_ENUM_VALUE(SOCKET_TYPE, type));
+        result = NULL;
+    }
+
+    else
+    {
+        // Codes_SOCKET_TRANSPORT_WIN32_09_002: [ socket_transport_create shall allocate a new SOCKET_TRANSPORT object. ]
+        result = malloc(sizeof(SOCKET_TRANSPORT));
+        if (result == NULL)
+        {
+            LogError("failure allocating SOCKET_TRANSPORT: %zu", sizeof(SOCKET_TRANSPORT));
+        }
+        else
+        {
+            // Codes_SOCKET_TRANSPORT_WIN32_09_003: [ socket_transport_create shall call sm_create to create a sm object. ]
+            result->sm = sm_create("Socket_transport_win32");
+            if (result->sm == NULL)
+            {
+                LogError("sm_create failed.");
+            }
+            else
+            {
+                result->type = type;
+                goto all_ok;
+            }
+            free(result);
+            // Codes_SOCKET_TRANSPORT_WIN32_09_004: [ On any failure socket_transport_create shall return NULL. ]
+            result = NULL;
+        }
+    }
+all_ok:
+    // Codes_SOCKET_TRANSPORT_WIN32_09_005: [ On success socket_transport_create shall return SOCKET_TRANSPORT_HANDLE. ]
+    return result;
+}
+
+void socket_transport_destroy(SOCKET_TRANSPORT_HANDLE socket_transport)
+{
+    // Codes_SOCKET_TRANSPORT_WIN32_09_006: [ If socket_transport is NULL socket_transport_destroy shall return. ]
+    if (socket_transport == NULL)
+    {
+        // Do nothing
+        return;
+    }
+    else
+    {
+        // Codes_SOCKET_TRANSPORT_WIN32_09_007: [ socket_transport_destroy shall call sm_destroy to destroy the sm object. ]
+        sm_destroy(socket_transport->sm);
+
+        // Codes_SOCKET_TRANSPORT_WIN32_09_008: [ socket_transport_destroy shall free the SOCKET_TRANSPORT_HANDLE object. ]
+        free(socket_transport);
+    }
+}
+
+
+int socket_transport_connect(SOCKET_TRANSPORT_HANDLE socket_transport, const char* hostname, uint16_t port, uint32_t connection_timeout_ms)
+{
+    int result;
+
+    // Codes_SOCKET_TRANSPORT_WIN32_09_009: [ If socket_transport is NULL, socket_transport_connect shall fail and return a non-zero value. ]
+    // Codes_SOCKET_TRANSPORT_WIN32_09_010: [If hostname is NULL, socket_transport_connect shall fail and return a non - zero value.]
+    // Codes_SOCKET_TRANSPORT_WIN32_09_011 : [If port is 0, socket_transport_connect shall fail and return a non - zero value.]
+    if (socket_transport == NULL ||
+        hostname == NULL ||
+        port == 0)
+    {
```

Comment:
Small comment: Usually we put the specs before each line of the NULL check.  So the hostname line would be between line 243 and 242